spec/index.adoc

```diff
@@ -4199,18 +4199,52 @@ Read response::
 * If `F(Q.Arg, Q.sender, Env) = Trap trap` then
 +
 ....
-{status: rejected; reject_code: CANISTER_ERROR, reject_message: <implementation-specific>, error_code: <implementation-specific>, node_signatures: <implementation-specific>}
+{status: "rejected"; reject_code: CANISTER_ERROR, reject_message: <implementation-specific>, error_code: <implementation-specific>, node_signatures: <implementation-specific>}
 ....
 * Else if `F(Q.Arg, Q.sender, Env) = Return {response = Reject (code, msg); …}` then
 +
 ....
-{status: rejected; reject_code: <code>: reject_message: <msg>, error_code: <implementation-specific>, node_signatures: <implementation-specific>}
+{status: "rejected"; reject_code: <code>: reject_message: <msg>, error_code: <implementation-specific>, node_signatures: <implementation-specific>}
 ....
 * Else if `F(Q.Arg, Q.sender, Env) = Return {response = Reply R; …}` then
 +
 ....
-{status: replied; reply: { arg :  <R> }, node_signatures: <implementation-specific>}
-....
+{status: "replied"; reply: {arg: <R>}, node_signatures: <implementation-specific>}
+....
+
+Given a certificate obtained by requesting the path `/subnet` via a read state request
```

Comment:
Could we somehow only read the keys we are interested in here as opposed to the entire `/subnet` path?

Reply:
That's tricky because we don't know the subnet ID of the canister based on the query response and thus cannot ask for the specific subtree of `/subnet` for the subnet hosting the queried canister.

Reply:
That's tricky. I also find that clients do not know the subnet id if they want to prefetch the node keys because they only know about the canister id.

Reply:
> I also find that clients do not know the subnet id if they want to prefetch the node keys because they only know about the canister id.

Note that based on what Martin is suggesting they wouldn't need to know the subnet ID. When making a read state query with the canister ID they are interested in then the request will end up on the right subnet. So when reading the entire `/subnet` subtree one would get the right keys even without knowing the subnet ID. One could then look up the subnet ID in the tree that comes back.